lib/c-api/src/wasm_c_api/module.rs

```diff
@@ -115,3 +115,364 @@ pub unsafe extern "C" fn wasm_module_serialize(
+                    assert(func_results->size == 0);
+                    wasm_valtype_vec_delete((wasm_valtype_vec_t*) func_results);
+                }
+
+                {
+                    wasm_exporttype_t* export_type = export_types.data[1];
+                    assert_exporttype_name(export_type, "global");
+
+                    const wasm_externtype_t* extern_type = wasm_exporttype_type(export_type);
+                    assert(wasm_externtype_kind(extern_type) == WASM_EXTERN_GLOBAL);
+
+                    wasm_globaltype_t* global_type = wasm_externtype_as_globaltype((wasm_externtype_t*) extern_type);
+                    assert(wasm_valtype_kind(wasm_globaltype_content(global_type)) == WASM_I32);
+                    assert(wasm_globaltype_mutability(global_type) == WASM_CONST);
+                }
+
+                {
+                    wasm_exporttype_t* export_type = export_types.data[2];
+                    assert_exporttype_name(export_type, "table");
+
+                    const wasm_externtype_t* extern_type = wasm_exporttype_type(export_type);
+                    assert(wasm_externtype_kind(extern_type) == WASM_EXTERN_TABLE);
+
+                    wasm_tabletype_t* table_type = wasm_externtype_as_tabletype((wasm_externtype_t*) extern_type);
+                    assert(wasm_valtype_kind(wasm_tabletype_element(table_type)) == WASM_FUNCREF);
+
+                    const wasm_limits_t* table_limits = wasm_tabletype_limits(table_type);
+                    assert(table_limits->min == 0);
+                    assert(table_limits->max == wasm_limits_max_default);
+                }
+
+                {
+                    wasm_exporttype_t* export_type = export_types.data[3];
+                    assert_exporttype_name(export_type, "memory");
+
+                    const wasm_externtype_t* extern_type = wasm_exporttype_type(export_type);
+                    assert(wasm_externtype_kind(extern_type) == WASM_EXTERN_MEMORY);
+
+                    wasm_memorytype_t* memory_type = wasm_externtype_as_memorytype((wasm_externtype_t*) extern_type);
+                    const wasm_limits_t* memory_limits = wasm_memorytype_limits(memory_type);
+                    assert(memory_limits->min == 1);
+                    assert(memory_limits->max == wasm_limits_max_default);
+                }
+
+                wasm_exporttype_vec_delete(&export_types);
+                wasm_byte_vec_delete(wasm);
+                wasm_module_delete(module);
+                wasm_store_delete(store);
+                wasm_engine_delete(engine);
+            }
+        })
+        .success();
+    }
+
+    #[test]
+    fn test_module_imports() {
+        (assert_c! {
+            #include <string.h>
+            #include "tests/wasmer_wasm.h"
+
+            void assert_importtype_name(const wasm_name_t* name, const char* expected) {
+                assert(strncmp(name->data, expected, strlen(expected)) == 0);
```

Comment:
```suggestion
                assert(strncmp(name->data, expected, name->size) == 0 && name->size == strlen(expected));
```
If `expected` is shorter than `name`, then the strncmp will stop at the NUL byte terminating `expected`, and the compare will fail, short-circuit evaluation skips the strlen part as the assert fires.
If `name` is shorter than `expected` and they match all the way through to name's end, the strlen will fail.

Reply:
Done, thanks.